### PR TITLE
CLC-5233 - E-Grades Export 2: Implement auto-download workflow

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.ets.berkeley.edu/jira/browse/CLC-5233

### CODE REVIEW

**Comment by redconfetti** [comment]
I forgot to decrease the cache time on this one. Do not merge.

File: src/assets/javascripts/angular/controllers/pages/canvasCourseGradeExportController.js [hunk 1]
```diff
@@ -17,7 +17,11 @@
      */
     $scope.goToGradebook = function() {
       var gradebookUrl = $scope.canvasRootUrl + '/courses/' + $scope.canvasCourseId + '/grades';
-      apiService.util.iframeParentLocation(gradebookUrl);
+      if (!!window.parent.frames.length) {
```
Comment (christianv) [inline src/assets/javascripts/angular/controllers/pages/canvasCourseGradeExportController.js]: @redconfetti can we use the `isInIframe` in the `util` service?

File: src/assets/javascripts/angular/controllers/pages/canvasCourseGradeExportController.js [hunk 2]
```diff
@@ -121,6 +127,34 @@
       return true;
     };
 
+    /*
+     * Switches to 'selection' step and scrolls to top of page
+     */
+    $scope.switchToSelection = function() {
+      // issue scroll to top based on Canvas or CalCentral context
+      if (!!window.parent.frames.length) {
```
Comment (christianv) [inline src/assets/javascripts/angular/controllers/pages/canvasCourseGradeExportController.js]: @redconfetti same here `isInIframe`

**Comment by raydavis** [comment]
Back-end looks good to me.

File: src/assets/javascripts/angular/controllers/pages/canvasCourseGradeExportController.js (hunk 1 above)
Comment (redconfetti) [inline src/assets/javascripts/angular/controllers/pages/canvasCourseGradeExportController.js]: @christianv I tried, but it wasn't working for me, but I see now why. It wasn't [exposed externally via the util service interface](https://github.com/ets-berkeley-edu/calcentral/blob/49ecbe9e/src/assets/javascripts/angular/services/utilService.js#L161). I'll add an update to fix this. Thanks for triggering this realization. :+1:

Comment (christianv) [inline src/assets/javascripts/angular/controllers/pages/canvasCourseGradeExportController.js]: @redconfetti thanks!

**Pull request merged**